generators/contract/templates/v2/private/typescript/src/my-contract.ts

```diff
@@ -74,14 +85,16 @@ export class <%= assetPascalCase %>Contract extends Contract {
         if (!exists) {
             throw new Error(`The asset <%= assetSpaceSeparator %> ${<%= assetCamelCase %>Id} does not exist`);
         }
+
+        const myCollectionName: string = await getCollectionName(ctx);
         await ctx.stub.deletePrivateData(myCollectionName, <%= assetCamelCase %>Id);
     }
 
     @Transaction()
-    public async verify<%= assetPascalCase %>(ctx: Context, <%= assetCamelCase %>Id: string, objectToVerify: <%= assetPascalCase %>): Promise<boolean> {
+    public async verify<%= assetPascalCase %>(ctx: Context, mspid: string, <%= assetCamelCase %>Id: string, objectToVerify: <%= assetPascalCase %>): Promise<boolean> {
         // Convert user provided object into a hash
         const hashToVerify: string = crypto.createHash('sha256').update(JSON.stringify(objectToVerify)).digest('hex');
-        const pdHashBytes: Uint8Array = await ctx.stub.getPrivateDataHash(myCollectionName, <%= assetCamelCase %>Id);
+        const pdHashBytes: Uint8Array = await ctx.stub.getPrivateDataHash(mspid, <%= assetCamelCase %>Id);
```

Comment:
The first argument in `getPrivateDataHash` expects a collection name, not mspid.
We should be doing:

```
const pdHashBytes: Uint8Array = await ctx.stub.getPrivateDataHash(`_implicit_org_${mspid}`, <%= assetCamelCase %>Id);
```

Or something along those lines.

Reply:
Ok, I'll change it to `collectionName: string` in the parameters and pass in `collectionName` to getPrivateDataHash.